File: book1_companion/psych_stats_with_python_companion_v0_1/scripts/r/run_all.R
#!/usr/bin/env Rscript
args <- commandArgs(trailingOnly = FALSE)
script_arg <- args[grep('^--file=', args)][1]
root <- normalizePath(file.path(dirname(sub('^--file=', '', script_arg)), '..', '..'), mustWork = TRUE)
source(file.path(root, "scripts", "r", "common.R"))

scripts <- c(
  "ch05_independent_t.R", "ch06_paired_t.R", "ch07_one_way_anova.R", "ch08_two_way_anova.R",
  "ch09_repeated_measures_anova.R", "ch10_correlation.R", "ch11_regression.R", "ch12_apa_reporting.R"
)
for (script in scripts) {
  status <- system2("Rscript", file.path(root, "scripts", "r", script))
  if (status != 0) stop(paste("failed:", script))
}

File: book1_companion/psych_stats_with_python_companion_v0_2_1/scripts/python/run_all.py
#!/usr/bin/env python3
from __future__ import annotations

import runpy
import sys
from pathlib import Path

HERE = Path(__file__).resolve().parent
SCRIPTS = [
    "ch05_independent_t.py",
    "ch06_paired_t.py",
    "ch07_one_way_anova.py",
    "ch08_two_way_anova.py",
    "ch09_repeated_measures_anova.py",
    "ch10_correlation.py",
    "ch11_regression.py",
    "ch12_apa_reporting.py",
]

if str(HERE) not in sys.path:
    sys.path.insert(0, str(HERE))
for name in SCRIPTS:
    runpy.run_path(str(HERE / name), run_name="__main__")
